src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -31,163 +31,136 @@ IConfiguration configuration
                 new AzureKeyCredential(_configuration["OpenAI:Key"]));
         }
 
-        public async Task<string> CreateDocumentAsync(string user, string badCode, string goodCode, string language, string comment, string[] guidelineIds)
+        public async Task<string> CreateDocumentAsync(string language, string badCode, string goodCode, string comment, string guidelineIds, string user)
         {
-            var embedding = await GetEmbeddingsAsync(badCode);
-
             var document = new CopilotCommentModel()
             {
+                Language = language,
                 BadCode = badCode,
                 GoodCode = goodCode,
-                Embedding = embedding,
-                Language = language,
+                Embedding = await GetEmbeddingsAsync(badCode),
                 Comment = comment,
-                GuidelineIds = guidelineIds,
+                GuidelineIds = SplitGuidelineIds(guidelineIds),
```

Comment:
Does the database not do this automatically?

Reply:
When I was testing with Postman, it sent the guideline ids as a string like `"id1, id2, id3"`. `SplitGuidelineIds` just parses it into an array, but if you know how to send an array with a REST API call, I will happily change it.